streaming/kafka/addons.tf

```diff
@@ -1,152 +1,97 @@
-    description = "The Amazon Elastic Block Store Container Storage Interface (CSI) Driver provides a CSI interface used by Container Orchestrators to manage the lifecycle of Amazon EBS volumes."
-    chart       = local.csi_name
-    version     = "2.12.1"
-    repository  = "https://kubernetes-sigs.github.io/aws-ebs-csi-driver"
-    namespace   = local.csi_namespace
-    values = [
-      <<-EOT
-      image:
-        repository: public.ecr.aws/ebs-csi-driver/aws-ebs-csi-driver
-      controller:
-        k8sTagClusterId: ${module.eks_blueprints.eks_cluster_id}
-      node:
-        tolerateAllTaints: true
-      EOT
-    ]
+module "ebs_csi_driver_irsa" {
+  source                = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version               = "~> 5.14"
+  role_name             = format("%s-%s", local.name, "ebs-csi-driver")
+  attach_ebs_csi_policy = true
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:ebs-csi-controller-sa"]
+    }
   }
+  tags = local.tags
+}
 
-  #---------------------------------------------------------------
-  # AWS Load Balancer Controller
-  #---------------------------------------------------------------
-  enable_aws_load_balancer_controller = true
-
-  #---------------------------------------------------------------
-  # Metrics Server
-  #---------------------------------------------------------------
-  enable_metrics_server = true
-  metrics_server_helm_config = {
-    name       = "metrics-server"
-    repository = "https://kubernetes-sigs.github.io/metrics-server/" # (Optional) Repository URL where to locate the requested chart.
-    chart      = "metrics-server"
-    version    = "3.8.2"
-    namespace  = "kube-system"
-    timeout    = "300"
-    values = [templatefile("${path.module}/helm-values/metrics-server-values.yaml", {
-      operating_system = "linux"
-      node_group_type  = "core"
-    })]
+#---------------------------------------------------------------
+# IRSA for VPC CNI
+#---------------------------------------------------------------
+module "vpc_cni_irsa" {
+  source                = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version               = "~> 5.14"
+  role_name             = format("%s-%s", local.name, "vpc-cni")
+  attach_vpc_cni_policy = true
+  vpc_cni_enable_ipv4   = true
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:aws-node"]
+    }
   }
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# Kubernetes Add-ons
+#---------------------------------------------------------------
+resource "aws_eks_addon" "vpc_cni" {
+  cluster_name = module.eks.cluster_name
+  addon_name   = "vpc-cni"
+  service_account_role_arn = module.vpc_cni_irsa.iam_role_arn
```

Comment:
add `preserve=true` to ensure VPC CNI is not fully deleted before the EKS Cluster. VPC CNI sometimes required to cleanup the existing add-ons

Reply:
Done